bindings/java/src/main/native/endpoint.cc

```diff
@@ -170,12 +191,21 @@ Java_org_openucx_jucx_ucp_UcpEndpoint_getNonBlockingNative(JNIEnv *env, jclass c
                                                            jlong size, jobject callback,
                                                            jint memory_type)
 {
-    ucs_status_ptr_t request = ucp_get_nb((ucp_ep_h)ep_ptr, (void *)laddr, size,
-                                          raddr, (ucp_rkey_h)rkey_ptr, jucx_request_callback);
+    ucp_request_param_t param = {0};
+
+    jobject jucx_request = jucx_request_allocate(env, callback, &param);
 
+    param.op_attr_mask |= UCP_OP_ATTR_FIELD_MEMORY_TYPE;
+    param.memory_type   = static_cast<ucs_memory_type_t>(memory_type);
+    param.cb.send       = jucx_request_callback;
```

Comment:
this blocks repeats in multiple functions, can you pls add helper function to avoid duplication?

Reply:
unhandled